tests/hipscat/pixel_tree/test_pixel_tree_builder.py

```diff
@@ -107,6 +201,70 @@ def test_pixel_builder_retrieve_none_node():
     assert tree_builder[HealpixPixel(10, 10)] is None
 
 
+def test_split_leaf_to_match_partitioning():
+    builder1 = PixelTreeBuilder()
+    builder2 = PixelTreeBuilder()
+    builder1.create_node((0, 0), PixelNodeType.LEAF)
+    builder2.create_node((0, 0), PixelNodeType.INNER)
+    builder2.create_node((1, 0), PixelNodeType.INNER)
+    builder2.create_node((2, 0), PixelNodeType.LEAF)
+    builder1.split_leaf_to_match_partitioning(builder2[(0, 0)])
+    assert builder1[(0, 0)].node_type == PixelNodeType.INNER
+    for pixel in [(1, 1), (1, 2), (1, 3)]:
+        assert pixel in builder1
+        assert builder1[pixel].node_type == PixelNodeType.LEAF
```

Comment:
To confirm my understanding pixel (1, 0) is also in builder1, but is an internal node and that is why it was left out of this check?

Reply:
Yeah, that's right, I've added a check for it separately as an inner node